internal/transport/transport.go

```diff
@@ -251,6 +252,10 @@ type Stream struct {
 
 	headerChan       chan struct{} // closed to indicate the end of header metadata.
 	headerChanClosed uint32        // set when headerChan is closed. Used to avoid closing headerChan multiple times.
+	headerValid      bool          // set if headerChan is closed due to valid
```

Comment:
Is this clearer?

`indicates whether a valid header was received. This field's value is only meaningful when headerChan is closed (always waitOnHeader() before reading it's valie)`

Reply:
I changed the wording a bit using your suggestion.